storybookjs/storybook
# NextJS: Add experimental RSC support

Closes #21540 

## What I did

Turns out that in NextJS's version of React, it's possible to render async components in the client using React Suspense.

```js
export default {
  component: MyServerComponent,
  decorators: [(Story) => <Suspense><Story /><Suspense />]
}
```

To make this ergonomic for NextJS users, I:

- [x] Added an experimental feature flag `experimentalNextRSC` to add RSC support
- [x] Added a parameter `nextjs.rsc` to selectively disable RSC on stories/components
- [x] Updated the README with basic documentation on how to use the feature

## Checklist for Contributors

### Testing



#### The changes in this PR are covered in the following automated tests:

#### Manual testing

Run the NextJS sandbox and view the sample stories. Observe that both the basic & nested stories work, and that the intentionally erroring stories error in reasonable ways.

### Documentation

      [MIGRATION.MD](https://github.com/storybookjs/storybook/blob/next/MIGRATION.md)

### 🦋 Canary release



This PR does not have a canary release associated. You can request a canary release of this pull request by mentioning the `@storybookjs/core` team here.

_core team members can create a canary release [here](https://github.com/storybookjs/storybook/actions/workflows/canary-release-pr.yml) or locally with `gh workflow run --repo storybookjs/storybook canary-release-pr.yml --field pr=<PR_NUMBER>`_
